src/main/java/com/extremelyd1/listener/ItemListener.java

```diff
@@ -53,4 +56,55 @@ public void onPlayerDropItem(PlayerDropItemEvent e) {
             e.setCancelled(true);
         }
     }
+
+    /**
+     * Caters for collecting milk, water or lava
+     *
+     * @param e - PlayerBucketFillEvent
+     */
+    @EventHandler
+    public void onBucketFill(PlayerBucketFillEvent e) {
+        if (!game.getState().equals(Game.State.IN_GAME)) {
+            e.setCancelled(true);
+            return;
+        }
+
+        game.onMaterialCollected(e.getPlayer(), e.getItemStack().getType());
+    }
+
+    /**
+     * This event caters for the collection of fishes into buckets, and the collection of mushroom stew from mooshrooms
+     *
+     * @param e - PlayerInteractAtEntityEvent
+     */
+    @EventHandler
+    public void onPlayerInteractAtEntity(PlayerInteractAtEntityEvent e) {
+        if (e.isCancelled()) return;
+
+        Map<EntityType, Material> fishes = new HashMap<EntityType, Material>() {{
+           put(EntityType.SALMON, Material.SALMON_BUCKET);
+           put(EntityType.COD, Material.COD_BUCKET);
+           put(EntityType.PUFFERFISH, Material.PUFFERFISH_BUCKET);
+           put(EntityType.TROPICAL_FISH, Material.TROPICAL_FISH_BUCKET);
+        }};
+
+        Map<EntityType, Material> foodHarvesting = new HashMap<EntityType, Material>() {{
+           put(EntityType.MUSHROOM_COW, Material.MUSHROOM_STEW);
+        }};
+
+        if (e.getPlayer().getInventory().getItemInMainHand().getType() == Material.WATER_BUCKET ||
+                e.getPlayer().getInventory().getItemInOffHand().getType() == Material.WATER_BUCKET) {
+            Material collectedMaterial = fishes.get(e.getRightClicked().getType());
+            if (collectedMaterial != null) game.onMaterialCollected(e.getPlayer(), collectedMaterial);
```

Comment:
Although there is no explicit coding style defined in this project, I would appreciate it if you could write out these one-line if-statements.

Reply:
I've expanded these out